src/docs.js

```diff
@@ -6,38 +6,48 @@ let path = require('path');
 let recursiveReadSync = require('recursive-readdir-sync');
 
 async function documenter(options) {
-
   options = _.defaults({}, options, {
+    metadata: null,
     schemas: [],
     docsFolder: null,
+    references: [],
   });
   assert(options.schemas, 'options.schemas must be given');
   assert(options.schemas instanceof Array, 'options.schemas must be an array');
 
-  let schemas = options.schemas;
   let tarball = tar.pack();
+  let metadata = options.metadata;
+
+  if (metadata) {
+    let data = JSON.stringify(metadata, null, 2);
+    tarball.entry({name: 'metadata.json'}, data);
```

Comment:
I think perhaps the best option/compromise here is to remove the ability to set arbitrary metadata and only have a `tier` argument to the library that is set in the metadata.

`version` will be set automatically. So inside the library it will basically be

```js
let metadata = {version: 1, tier: options.tier};
```

and `tier` will have to be ensured to be either one of `core` or `platform`.